lib/nostrum/cache/guild/guild_server.ex

```diff
@@ -442,4 +447,14 @@ defmodule Nostrum.Cache.Guild.GuildServer do
     {:reply, {guild_id, old_emojis, emojis}, %{state | emojis: emojis}}
   end
 
+  def handle_call({:update, :voice_state, guild_id, voice_state}, _from, state) do
+    old_voice_states = state.voice_states
+    new_voice_states =
+      [
+        voice_state |
+        state.voice_states
+        |> Enum.reject(fn(map) -> map.user_id == voice_state.user_id end)
+      ]
+    {:reply, {guild_id, old_voice_states, new_voice_states}, %{state | voice_states: new_voice_states}}
```

Comment:
Yes, I agree with that, and in that point of view, I'm very questioning about receiving new/old lists *nor* new/old states. What will be the usages (if there are other ways to go)? So far I can't see any of it, so maybe we could consider it later; I'll go leave this as it is, just for now.